### PR TITLE
[labs/context] Make @consume decorator work with optional fields

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes https://github.com/lit/lit/issues/3314

This doesn't allow `@provide()` to work with optional fields, since if you provide a value for a non-nullable context you should not use optional fields.

cc @benjamind

### CODE REVIEW

File: packages/labs/context/src/lib/decorators/consume.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export function consume<ValueType>({
   context: ContextKey<unknown, ValueType>;
   subscribe?: boolean;
 }): <K extends PropertyKey>(
-  protoOrDescriptor: ReactiveElement & Record<K, ValueType>,
+  protoOrDescriptor: ReactiveElement,
```
Comment (kevinpschaaf) [inline packages/labs/context/src/lib/decorators/consume.ts:51]: Seems like the generics `ValueType` and `K` are no longer used, remove?

Comment (kevinpschaaf) [reply packages/labs/context/src/lib/decorators/consume.ts:51]: I guess this just means we remove what little type safety existed... but I guess it's just possible to express?

Does `Partial<Record<K, ValueType>>` help? That's sorta like _if_ it's there then it has to have the right type?

Comment (justinfagnani) [reply packages/labs/context/src/lib/decorators/consume.ts:51]: Yeah, that's better. I added a test that we get the error too.

File: packages/labs/context/src/lib/decorators/consume.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export function consume<ValueType>({
   context: ContextKey<unknown, ValueType>;
   subscribe?: boolean;
 }): <K extends PropertyKey>(
-  protoOrDescriptor: ReactiveElement & Record<K, ValueType>,
+  protoOrDescriptor: ReactiveElement & Partial<Record<K, ValueType>>,
```
Comment (kevinpschaaf) [inline packages/labs/context/src/lib/decorators/consume.ts:51]:
```suggestion
  // Partial<> allows for providing the value to an optional field
  protoOrDescriptor: ReactiveElement & Partial<Record<K, ValueType>>,
```

**Review by kevinpschaaf — APPROVED** [review]
LGTM, let's add a comment so we remember why we did this

**Pull request merged**